mbedtls/src/pk/mod.rs

```diff
@@ -1564,4 +1625,37 @@ iy6KC991zzvaWY/Ys+q/84Afqa+0qJKQnPuy/7F5GkVdQA/lfbhi
             assert_eq!(l.unwrap(), LEN);
         }
     }
+
+    #[test]
+    fn private_from_rsa_components_sanity() {
+        let mut pk = Pk::generate_rsa(&mut crate::test_support::rand::test_rng(), 2048, 0x10001).unwrap();
+        let components = RsaPrivateComponents::WithPrimes {
+            p: &pk.rsa_private_prime1().unwrap(),
+            q: &pk.rsa_private_prime2().unwrap(),
+            e: &Mpi::new(pk.rsa_public_exponent().unwrap() as _).unwrap(),
+        };
+        let mut pk2 = Pk::private_from_rsa_components(components).unwrap();
+        assert_eq!(pk.write_private_der_vec().unwrap(), pk2.write_private_der_vec().unwrap());
+
+        let components = RsaPrivateComponents::WithPrivateExponent {
+            n: &pk.rsa_public_modulus().unwrap(),
+            d: &pk.rsa_private_exponent().unwrap(),
+            e: &Mpi::new(pk.rsa_public_exponent().unwrap() as _).unwrap(),
+        };
+        let mut pk3 = Pk::private_from_rsa_components(components).unwrap();
+        assert_eq!(pk.write_private_der_vec().unwrap(), pk3.write_private_der_vec().unwrap());
```

Comment:
This test fails when I don't use the test `rng`:
```
        use crate::rng::{CtrDrbg, OsEntropy};
        let entropy = OsEntropy::new();
        let mut rng = CtrDrbg::new(entropy.into(), None).unwrap();
        let mut pk = Pk::generate_rsa(&mut rng, 2048, 0x10001).unwrap();
```

It fails in the last assertion. I think it swaps parameter order on DER serialization?

Reply:
Perhaps we should refactor the test RNG so that it randomizes tests, or have another test RNG. (cc. @jethrogb)